lib/components/particle_component.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:ui';
+
+import 'package:flame/components/particles/composed_particle.dart';
+import 'package:flutter/foundation.dart';
+
+import '../time.dart';
+import 'component.dart';
+
+/// A function which returns [Particle] when called
+typedef ParticleGenerator = Particle Function(int);
+
+/// Base class implementing common behavior for all the particles.
+///
+/// Intention is to follow same "Extreme Composability" style
+/// as across the whole Flutter framework, so each type of particle implements
+/// some particular behavior which then could be nested and combined together
+/// to create specifically required experience.
+abstract class Particle extends Component {
```

Comment:
Since this extends from component, I would call it `ParticleComponent` so it is explicit that this is class to be used on Flame's component system

Reply:
Fixed as per our other discussion, `Particle` is now independent from `Component`